src/data_loader/osm.py

```diff
@@ -9,17 +9,20 @@
 OSM_DIR = DATA_DIR / "osm_graph"
 
 
-def download_osm_graph(bbox: tuple[float, float, float, float]) -> nx.MultiDiGraph:
+def download_osm_graph(
+    bbox: tuple[float, float, float, float], simplify: bool = True
+) -> nx.MultiDiGraph:
     """
     Download an OSM graph from a bounding box.
     :param bbox: Bounding box of the area to download, formatted (left, bottom, right, top).
+    :param simplify: Whether to simplify the graph.
     :return: OSM graph containing the road network in the bounding box.
     """
     logging.info("Downloading OSM graph")
     city_graph = ox.graph_from_bbox(
         bbox=bbox,
         network_type="drive_service",
-        retain_all=True,
```

Comment:
What was the argument here? We do not want disconnected components, because we assume there aren't any?

Reply:
I'm open to do either, but I guess it's mainly just increasing memory usage and computational time without adding value to the shortest path calculations. If components are disconnected from the rest of the graph, we won't be able to find valid paths from/through/to them.

Reply:
Agreed.